src/utils/addresses.ts

```diff
@@ -52,3 +52,19 @@ export const parsePrefixedAddress = (value: string): PrefixedAddress => {
 export const formatPrefixedAddress = (address: string, prefix?: string): string => {
   return prefix ? `${prefix}:${address}` : address
 }
+
+export const cleanInputValue = (value: string): string => {
+  const regex = /(0x[a-fA-F0-9]{40})\b/
+  const regexWithPrefix = /([a-zA-Z0-9]+):(0x[a-fA-F0-9]{40})\b/
```

Comment:
You could do it with one regex like so:
```
/(?:([a-z0-9]+):)?(0x[a-f0-9]{40})\b/i
```

<img width="868" alt="Screenshot 2023-05-09 at 06 44 03" src="https://user-images.githubusercontent.com/381895/236996131-042425b9-e073-48d3-843d-bd9deb873cd2.png">

<img width="824" alt="Screenshot 2023-05-09 at 06 46 16" src="https://user-images.githubusercontent.com/381895/236996452-a2d688a5-44d1-4ddb-97ff-de032196f661.png">

Also you could combine test and match in one call:
```
const match = addressRe.match(value)
if (match) return match[0] // address with optional prefix
```

Reply:
updated logic to use single regex